app/controllers/main_controller.rb

```diff
@@ -1,4 +1,5 @@
 class MainController < ApplicationController
+  before_action :authenticate_user!, only: [:home]
```

Comment:
By using `only: [:home]` you are saying that authentication is only required in this action, in other words, authentication is something specific to this action.

This is a bit risky, we are increasing the chances of leak actions that should be authenticated because we are requiring us and other future developers to remember that always we need to add a before action like this for authenticated actions.

I propose we oppositely do that.
All the actions in this application will be authenticated by default, if we want to add an unauthenticated action we need to state that explicitly.

So my suggestion is that we add that `before_action :authenticate_user!` in the `ApplicationController`, this way all controllers will have their actions authenticated by default.

If we want to add, for example an unauthenticated page to this app, we should do something like this:

```ruby
class PublicPagesController < ApplicationController
  before_action :authenticate_user!, except: [:index]

  def index
  end
end
```

Reply:
Just leaving here a good tutorial so that you can check again what we have done with devise here: https://www.digitalocean.com/community/tutorials/how-to-set-up-user-authentication-with-devise-in-a-rails-7-application

Reply:
Hello americo, 

Okay i am going to make those changes right away, i went through the tutorial and i understand what you are saying, but in what case do we use only:

Reply:
One thing first: I just updated my first message: after adding `before_filter :authenticate_user!` to `ApplicationController` you should use `skip_before_action :authenticate_user!, only: [:index]` if you want to skip certain `:index`.

Now let me try to answer your question: but in what case do we use only?

We use `only` in cases where the default is doing nothing and we want to do something for specific actions. It is like you are telling others that if they want to reuse that piece of functionality they need to be explicit.

One very common example (but not much valuable) is in CRUDs:

```ruby
class PostsController < ApplicationController
  before_filter :find_post, only: %i[show update destroy]

  def index
    @post = Post.all
  end

  def create
    @post = Post.create(post_params)
  end

  def show
  end

  def update
    @post.update(post_params)
  end

  def destroy
    @post.destroy
  end

  private def find_post
    @post = Post.find(params[:id])
  end

  private def post_params
    params.require(:post).permit(:title, :content)
  end
end
```

As you can see, we don't want if someone to add a new action to `PostsController`, that action automatically have a `@post` defined, because maybe that action is not about `@post` but is about `@posts`. So the person needs to explicitly add the new action to the `only` list if they want that `before_filter` for the new action.

Does that make sense to you?